packages/frontend/src/contexts/state.js

```diff
@@ -1,12 +1,15 @@
 import { createContext } from 'react'
 import Info from './info'
+import Interface from './interface'
 
 const state = {}
 
 const info = new Info(state)
+const ui = new Interface()
 
 Object.assign(state, {
   info,
+  ui,
 })
```

Comment:
@vimwitch 
> This file shouldn't export a context, it should export the class above.
> 
> I recently changed this pattern so that we have a single state object that exports each store. This allows the stores to interact with each other without having to circularly import each other.

I have assigned `info` and `ui` to `state` before `createContext(state)`

then  `const { info } = useContext(state)` is used in Home.jsx,
and `const { ui } = React.useContext(state)` in Tooltip.jsx

It appears to work, but did I understand you correctly?

Reply:
Yep, make sure to pass the `state` object to the `Interface` constructor [here](https://github.com/Unirep/explorer/pull/3/files/922203509c648ff4d98ffc446dfa283a7a14d6af..8a813897163cab86037208d442b612431b08da51#diff-50f48855fdb449ec98bd0631d2dcd55cacc351ce396fcdee8948867b2a568b93R8). Even if it's not used it keeps the pattern consistent.

Reply:
Sure, I left some review notes on the PR. If you're taking TS out make sure to pull the config out too, otherwise we're preprocessing for nothing.

The PR looks good to me. I'm not worried about the ternary or style stuff yet, just need a good scaffold to build on.

Reply:
We can make issues for this project if you like. It will probably be like per page. "Ingest the data for the homepage" "Add support for retrieving attestations by attester" stuff like that. Sorta up to you, but when I start a project like this I like to build against the designs directly because issues become something else to maintain.